src/kontrol/foundry.py

```diff
@@ -542,6 +576,14 @@ def latest_proof_version(
         versions = {int(pid.split(':')[1]) for pid in proof_ids}
         return max(versions, default=None)
 
+    def latest_proof_version_before(self, test: str, base_test_ver: int) -> int | None:
+        """
+        find the highest used proof ID, less than. Returns None if no version of this proof exists.
```

Comment:
Some nitpicking on [docstrings](https://peps.python.org/pep-0257/#multi-line-docstrings):
```suggestion
        Find the highest used proof ID, less than base_test_ver. 
        
        Returns None if no version of this proof exists.
```